mapie/quantile_regression.py

```diff
@@ -0,0 +1,425 @@
+        "estimators_",
+        "conformity_scores_",
+        "n_samples_calib",
+        "n_features_in_"
+    ]
+
+    Params = TypedDict(
+        "Params",
+        {
+            "loss_name": str,
+            "alpha_name": str
+        },
+    )
+    quantile_estimator_params = {
+        "GradientBoostingRegressor": Params(
+            loss_name="loss",
+            alpha_name="alpha"
+        ),
+        "QuantileRegressor": Params(
+            loss_name="quantile",
+            alpha_name="quantile"
+        ),
+        "HistGradientBoostingRegressor": Params(
+            loss_name="loss",
+            alpha_name="alpha"
+        ),
+        "LGBMRegressor": Params(
+            loss_name="objective",
+            alpha_name="alpha"
+        )
+    }
+
+    def __init__(
+        self,
+        estimator: Optional[Union[RegressorMixin, Pipeline]] = None,
+        method: str = "quantile",
+        alpha: float = 0.1,
+        cv: Optional[Union[int, str]] = None,
+    ) -> None:
+        super().__init__(
+            estimator=estimator,
+            method=method,
+            cv=cv
+        )
+        self.alpha = alpha
+
+    def _check_alpha(
+        self,
+        alpha: float,
+    ) -> NDArray:
+        """
+        Perform several checks on the alpha value and changes it from
+        a float to an ArrayLike
+
+        Parameters
+        ----------
+        alpha : float
+            Can only be a float value between 0 and 0.5.
+            Represent the uncertainty of the confidence interval.
+            Lower alpha produce larger (more conservative) prediction
+            intervals. Alpha is the complement of the target coverage level.
+            Only used at prediction time. By default 0.1
+
+        Returns
+        -------
+        ArrayLike
+            An ArrayLike of three values, first one being the lower quantile
+            value, second the upper quantile value and final being the quantile
+            at which the prediction will be made.
+
+        Raises
+        ------
+        ValueError
+            If alpha is not a float.
+        ValueError
+            If the value of alpha is not between 0 and 0.5.
+        """
+        if isinstance(alpha, float):
+            if np.any(np.logical_or(alpha <= 0, alpha >= 0.5)):
+                raise ValueError(
+                    "Invalid alpha. Allowed values are between 0 and 0.5."
+                )
+            else:
+                alpha_np = np.array([alpha / 2, 1 - alpha / 2, 0.5])
+        else:
+            raise ValueError(
+                "Invalid alpha. Allowed values are float."
+            )
+        return alpha_np
+
+    def _check_estimator(
+        self,
+        estimator: Optional[Union[RegressorMixin, Pipeline]] = None
+    ) -> Union[RegressorMixin, Pipeline]:
+        """
+        Perform several checks on the estimator to check if it has
+        all the required specifications to be used with this methodology.
```

Comment:
Be more explicit for "this methodology" (describe in a few words why this method performs all these checks).